genswagger/genswagger.go

```diff
@@ -0,0 +1,284 @@
+package genswagger
+
+import (
+	"github.com/TuneLab/go-truss/gendoc/doctree"
+
+	swagger "github.com/go-openapi/spec"
+	"github.com/pkg/errors"
+	//spew "github.com/davecgh/go-spew/spew"
+
+	"github.com/go-openapi/loads"
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/validate"
+
+	"github.com/TuneLab/go-truss/truss/truss"
+)
+
+type ParamsField struct {
+	*doctree.HttpParameter
+	MethodField *doctree.MessageField
+}
+
+func GenerateSwaggerFile(dt doctree.Doctree) ([]truss.NamedReadWriter, error) {
+	svc, err := getProtoService(dt)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get service from doctree")
+	}
+
+	spec := genSwagObject(svc)
+
+	swaggerBytes, err := spec.MarshalJSON()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to marshal swagger to json")
+	}
+
+	var file truss.SimpleFile
+	file.Path = "service/swagger/swagger.json"
+	file.Write(swaggerBytes)
+
+	var files []truss.NamedReadWriter
+	files = append(files, &file)
+
+	return files, nil
+}
+
+func validateSwaggerBytes(spec []byte) error {
+	doc, err := loads.Analyzed(spec, "2.0")
+	if err != nil {
+		return errors.Wrap(err, "swagger file can not be loaded into swagger doc")
+	}
+
+	err = validate.Spec(doc, strfmt.Default)
+	if err != nil {
+		return errors.Wrap(err, "swagger file cannot be validated")
+	}
+
+	return nil
+}
+
+// Spec: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#swagger-object
```

Comment:
This comment applies to most all of the functions related to swagger generation:

Though you have technically provided a comment for most of these functions, I think adding explanations of what these function do, how each maps from `doctree` structs to swagger structs and the particularities of that mapping would make it much more clear what this package is doing and *why*.

With the only comments being links to the spec, you imply the "what", but the "how" and "why" is left unaddressed.